runtime/common/ArgumentConversion.h

```diff
@@ -0,0 +1,51 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#pragma once
+
+#include "cudaq/Optimizer/Dialect/CC/CCOps.h"
+#include "cudaq/Optimizer/Dialect/CC/CCTypes.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/Types.h"
+
+namespace cudaq {
+class state;
+}
+
+namespace cudaq::opt {
+
+class ArgumentConverter {
+public:
+  /// Build an instance to create argument substitutions for a specified \p
+  /// kernelName in \p sourceModule.
+  ArgumentConverter(mlir::StringRef kernelName, mlir::ModuleOp sourceModule);
+
+  /// Generate a substitution ModuleOp for the vector of arguments presented.
+  /// The arguments are those presented to the kernel, kernelName.
+  void gen(const std::vector<void *> &arguments);
+
+  /// Get the list of substitutions that were generated by `gen()`.
+  mlir::SmallVector<cudaq::cc::ArgumentSubstitutionOp> &getSubstitutions() {
+    return substitutions;
+  }
+
+  /// Some substitutions may generate global constant information. Use this
+  /// interface to access both the substitutions and any global constants
+  /// created.
+  mlir::ModuleOp getSubstitutionModule() { return substModule; }
+
+private:
+  mlir::ModuleOp sourceModule;
+  mlir::ModuleOp substModule;
+  mlir::OpBuilder builder;
+  mlir::StringRef kernelName;
+
+  mlir::SmallVector<cc::ArgumentSubstitutionOp> substitutions;
```

Comment:
Is the extra line intentional?
```suggestion
  mlir::StringRef kernelName;
  mlir::SmallVector<cc::ArgumentSubstitutionOp> substitutions;
```